components/producers/github-codeql/main.go

```diff
@@ -0,0 +1,179 @@
+	"log"
+	"log/slog"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/google/go-github/v65/github"
+
+	v1proto "github.com/ocurity/dracon/api/proto/v1"
+	"github.com/ocurity/dracon/components/producers"
+	wrapper "github.com/ocurity/dracon/pkg/github"
+)
+
+type config struct {
+	// RepositoryOwner is the owner of the GitHub repository
+	RepositoryOwner string
+
+	// RepositoryName is the name of the GitHub repository
+	RepositoryName string
+
+	// GitHubToken is the GitHub token used to authenticate
+	GitHubToken string
+
+	// Ref is the Ref/branch to get alerts for
+	Ref string
+
+	// Severity if specified, only code scanning alerts with this severity will be returned. Possible values are: critical, high, medium, low, warning, note, error
+	Severity string
+
+	// toolName is internal
+	toolName string
+}
+
+var cfg config
+
+func main() {
+	flag.StringVar(&cfg.RepositoryOwner, "repository-owner", "", "The owner of the GitHub repository")
+	flag.StringVar(&cfg.RepositoryName, "repository-name", "", "The name of the GitHub repository")
+	flag.StringVar(&cfg.GitHubToken, "github-token", "", "The GitHub token used to authenticate with the API")
+	flag.StringVar(&cfg.Ref, "reference", "", "The Ref/branch to get alerts for")
+	flag.StringVar(&cfg.Severity, "severity", "", "If specified, only code scanning alerts with this severity will be returned. Possible values are: critical, high, medium, low, warning, note, error")
+	cfg.toolName = "CodeQL"
+	if err := producers.ParseFlags(); err != nil {
+		log.Fatal(err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	apiClient := wrapper.NewClient(cfg.GitHubToken)
+	alerts, err := listAlertsForRepo(ctx, apiClient, cfg.RepositoryOwner, cfg.RepositoryName, cfg.toolName, cfg.Severity)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	issues := parseIssues(alerts)
+
+	if err := producers.WriteDraconOut(
+		"github-code-scanning",
+		issues,
+	); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func doCall(ctx context.Context, apiClient wrapper.Wrapper, owner, repo string, opt *github.AlertListOptions) ([]*github.Alert, *github.Response, error) {
+	alerts, resp, err := apiClient.ListRepoAlerts(ctx, owner, repo, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if resp.StatusCode > 299 {
+		defer resp.Body.Close()
+		body, bodyReadErr := io.ReadAll(resp.Body)
+		if bodyReadErr != nil {
+			return nil, nil, errors.Errorf("could not read response error from github, err:%w", bodyReadErr)
+		}
+		slog.Error("github response", slog.String("body", string(body)))
+		return nil, nil, errors.Errorf("did not receive a valid status code from github, status code: %d, body:%s", resp.StatusCode, string(body))
+	}
+	return alerts, resp, nil
+}
+
+func listAlertsForRepo(ctx context.Context, apiClient wrapper.Wrapper, owner, repo, toolName, severity string) ([]*github.Alert, error) {
+	var allAlerts []*github.Alert
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
```

Comment:
It would be nice to make this configurable via env var `GITHUB_CLIENT_REQUEST_TIMEOUT=5m`.

It would need to be parsed like https://github.com/ndisidore/go-env/blob/main/parser.go#L71